### ISSUE TITLE
Power Advantage drain + EnderIO fluid tank crash

### ISSUE THREAD

**Issue by OhiraKyou** [cyanobacterium/PowerAdvantageAPI#24]
Sucking water from a Power Advantage drain into an EnderIO fluid tank causes a crash then and on subsequent world loading attempts.

I am also posting this report to the EnderIO issues.
### Data

[Crash log](http://pastebin.com/JvnmmdZp)
[Screenshots with steps](http://imgur.com/a/Eycct)
[Example world that crashes on load (489 KB)](http://www.mediafire.com/download/nki8p2l27t3yq5m)
### Versions

**Minecraft**
- Minecraft 1.9.4
- Forge 1.9.4-12.17.0.1951

**EnderIO**
- EnderIO 1.9.4-3.0.0.37_alpha
- EnderCore 1.9.4-0.4.0.23

**Power Advantage**
- OreSpawn 1.9.4-1.0.2
- BaseMetals 1.9.4-2.3.2.6
- PowerAdvantage 1.9.4-2.2.1

**Comment by HenryLoenwind** [comment]
oh yes, I see what's happening here:

```
    if(getTank().getFluidAmount() > 0 && getTank().getFluid().getFluid().equals(fluid.getFluid())){
        return getTank().drain(fluid.amount,forReal);
    } else {
        return new FluidStack(getTank().getFluid().getFluid(),0);
    }
```

That "else" won't work that well for an empty tank.

**Comment by DrPlantabyte** [comment]
Oops. I'll fix that soon. Thanks for the catch, HenryLoenwind. I wish fluid tanks were a little more deterministic in their behavior ("empty" tanks often, but not always, have a fluid stack) and didn't require so many null checks.

**Comment by HenryLoenwind** [comment]
Yes, those tank classes are not the best piece of software. In Ender IO we are subclassing them to make life a bit easier.

**Comment by DrPlantabyte** [comment]
Now fixed in version 2.2.2 (I think).
